src/schemas/dataset-schema.js

```diff
@@ -74,6 +74,12 @@ class DatasetSchema extends Schema {
   key = 'dataset';
 
   save(dataset) {
+    if (dataset.dataContainer) {
+      dataset = {
+        ...dataset,
```

Comment:
`dataset` is a class, this will make it a plain object. you can use the `copyTableAndUpdate` from `kepler-table.js` to avoid directly modifying `dataset